src/LeanMapper/Repository.php

```diff
@@ -263,15 +264,28 @@ protected function persistHasManyChanges(Entity $entity)
                         ];
                     }
                 } else {
-                    $this->connection->query(
-                        'DELETE FROM %n WHERE %n = ? AND %n = ? %lmt',
-                        $relationshipTable,
-                        $columnReferencingSourceTable,
-                        $entity->$idField,
-                        $columnReferencingTargetTable,
-                        $value,
-                        -$count
-                    );
+                    if ($driver instanceof \Dibi\Drivers\PostgreDriver) {
+                        $this->connection->query(
+                            'DELETE FROM %n WHERE [ctid] IN (SELECT [ctid] FROM %n WHERE %n = ? AND %n = ? LIMIT %i)',
+                            $relationshipTable,
+                            $relationshipTable,
```

Comment:
neni tu ten radek omylem 2x?
Pokud se meni jen sql prikaz (query), tak prosim o upraveni podminky tak, aby volani `$this->connection->query($query...` bylo jen jedno

Reply:
Bohužel není - jméno tabulky je nutné předat 2x - jednou pro `DELETE FROM` a podruhé pro subselect `SELECT ... FROM`, jinak by stačilo jen přepsat samotné query, ale tady se mění i parametry. Alternativně by to šlo přepsat ještě do tohoto tvaru:

``` php
$query = array();

if ($driver instanceof \Dibi\Drivers\PostgreDriver) {
	$query[] = 'DELETE FROM %n WHERE [ctid] IN (SELECT [ctid] FROM %n WHERE %n = ? AND %n = ? LIMIT %i)';
	$query[] = $relationshipTable;

} else {
	$query[] = 'DELETE FROM %n WHERE %n = ? AND %n = ? %lmt';
}

$query[] = $relationshipTable;
$query[] = $columnReferencingSourceTable;
$query[] = $entity->$idField;
$query[] = $columnReferencingTargetTable;
$query[] = $value;
$query[] = -$count;

$this->connection->query($query);
```

Případně parametry přesunout do pole a pro PostgreSQL udělat `array_unshift($params, $relationshipTable)` a vypodmínkovat query.